<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>743</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>NYCU PP25 1st Group Final Project</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="groupBox_db">
      <property name="title">
       <string>Database Operations</string>
      </property>
      <layout class="QHBoxLayout" name="dbLayout">
       <item>
        <widget class="QPushButton" name="btnCreateDB">
         <property name="text">
          <string>Create DB</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnOpenDB">
         <property name="text">
          <string>Open DB</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="labelDBInfo">
      <property name="text">
       <string>Database: (none)</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="btnShowSchema">
      <property name="text">
       <string>Show DB Items</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_insert">
      <property name="title">
       <string>Insert &amp; Search</string>
      </property>
      <layout class="QVBoxLayout" name="insertSearchLayout">
       <item>
        <widget class="QLineEdit" name="lineEdit">
         <property name="placeholderText">
          <string>Search by image name...</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="buttonLayout">
         <item>
          <widget class="QPushButton" name="btnInsert">
           <property name="text">
            <string>Insert</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btnOpenImage">
           <property name="text">
            <string>Open Image</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btnSearch">
           <property name="text">
            <string>Search</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btnFindSimilar">
           <property name="text">
            <string>Find Similar</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="HD">
           <property name="text">
            <string>Hamming Distance</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBox"/>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>55</height>
       </size>
      </property>
      <property name="title">
       <string>Compare Images(Pixel Level)</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QRadioButton" name="radioQtCompare">
         <property name="text">
          <string>Pixel Match</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">buttonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelTimeQt">
         <property name="text">
          <string>Pixel Match Time: ms</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioQtOpenMP">
         <property name="text">
          <string>Pixel Match (OpenMP)</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">buttonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelTimeQtOpenMP">
         <property name="text">
          <string>Pixel Match (OpenMP) Time: ms</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="btnRun">
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>55</height>
       </size>
      </property>
      <property name="baseSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string> Matching Method</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="labelRepeat">
         <property name="text">
          <string>Repeat Count:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinRepeat">
         <property name="suffix">
          <string/>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioWhileLoop">
         <property name="layoutDirection">
          <enum>Qt::LayoutDirection::LeftToRight</enum>
         </property>
         <property name="text">
          <string>Sequential</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">buttonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioOpenMP">
         <property name="text">
          <string>OpenMP</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">buttonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioPthread">
         <property name="text">
          <string>PThread</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">buttonGroup</string>
         </attribute>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="TimeLabel">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>55</height>
       </size>
      </property>
      <property name="title">
       <string>Time Comparison</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLabel" name="labelTimeSequential">
         <property name="text">
          <string>Sequential Time: ms</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelTimeOpenMP">
         <property name="text">
          <string>OpenMP Time: ms</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelTimePThread">
         <property name="text">
          <string>PThread Time: ms</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_list">
      <property name="title">
       <string>Image List</string>
      </property>
      <layout class="QVBoxLayout" name="listLayout">
       <item>
        <widget class="QListWidget" name="listWidget"/>
       </item>
       <item>
        <widget class="QLabel" name="imageLabel">
         <property name="minimumSize">
          <size>
           <width>300</width>
           <height>200</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::Shape::Box</enum>
         </property>
         <property name="text">
          <string>(image preview)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_actions">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>55</height>
       </size>
      </property>
      <property name="title">
       <string>Actions</string>
      </property>
      <layout class="QHBoxLayout" name="actionLayout">
       <item>
        <widget class="QPushButton" name="btnEdit">
         <property name="text">
          <string>Edit</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnDelete">
         <property name="text">
          <string>Delete</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
